contrib/unique_ptr/unique_ptr.hpp

```diff
@@ -462,18 +471,27 @@ class unique_ptr<T[], D>
         return tmp;
     }
 
-    void swap(unique_ptr& u) {detail_unique_ptr::swap(ptr_, u.ptr_);}
+    void swap(unique_ptr& u) {boost::detail_unique_ptr::swap(ptr_, u.ptr_);}
 private:
     template <class U>
         explicit unique_ptr(U,
-            typename enable_if_c<detail_unique_ptr::is_convertible<U, pointer>::value>::type* = 0);
+                            typename boost::enable_if_c<boost::detail_unique_ptr::is_convertible<U, pointer>::value>::type* = 0);
 
     template <class U>
-        unique_ptr(U, typename mpl::if_<is_reference<D>,
-                          volatile typename remove_reference<D>::type&, D>::type,
-                          typename enable_if_c<detail_unique_ptr::is_convertible<U, pointer>::value>::type* = 0);
+    unique_ptr(U, typename boost::mpl::if_<boost::is_reference<D>,
+               volatile typename boost::remove_reference<D>::type&, D>::type,
+               typename boost::enable_if_c<boost::detail_unique_ptr::is_convertible<U, pointer>::value>::type* = 0);
 };
 
+#ifdef LIBMESH_IS_COMPILING_HINNANT_UNIQUE_PTR
+} // namespace libMesh
+
+
+namespace boost {
+
+using libMesh::unique_ptr;
```

Comment:
Why declare it in libMesh and then using-declare it in boost, rather than the other way around?